docs/.vitepress/theme/components/WhyRolldownModal.vue

```diff
@@ -0,0 +1,159 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+import { onKeyStroke, } from '@vueuse/core';
+
+const VIDEO_ID = 'RRjfm8cMveQ';
+
+const emit = defineEmits(['close']);
+
+const isModalVisible = ref(false);
+
+// Scroll lock
+watch(isModalVisible, (value) => {
+  value
+    ? (document.documentElement.style.overflow = 'hidden')
+    : (document.documentElement.style.overflow = 'auto');
+},
+  { immediate: true }
+);
+
+const openModal = () => {
+  isModalVisible.value = true;
+};
+
+const closeModal = () => {
+  isModalVisible.value = false;
+  emit('close');
+};
+
+onKeyStroke('Escape', () => {
+  if (isModalVisible.value) {
+    closeModal();
+  }
+});
+</script>
+
+<template>
+  <button @click="openModal" class="open-modal-button">
+    Why Rolldown?
+    <svg class="icon-play" aria-labelledby="simpleicons-play-icon" role="img" viewBox="0 0 100 125" fill="#FFFFFF">
+      <title id="simpleicons-play-icon" lang="en" data-v-bf2d099b="">Play icon</title>
+      <path
+        d="M50,3.8C24.5,3.8,3.8,24.5,3.8,50S24.5,96.2,50,96.2S96.2,75.5,96.2,50S75.5,3.8,50,3.8z M71.2,53.3l-30.8,18  c-0.6,0.4-1.3,0.5-1.9,0.5c-0.6,0-1.3-0.1-1.9-0.5c-1.2-0.6-1.9-1.9-1.9-3.3V32c0-1.4,0.8-2.7,1.9-3.3c1.2-0.6,2.7-0.6,3.8,0  l30.8,18c1.2,0.6,1.9,1.9,1.9,3.3S72.3,52.7,71.2,53.3z"
+        data-v-bf2d099b=""></path>
+    </svg>
+  </button>
+  <Teleport to="body" v-if="isModalVisible">
+    <transition name="modal-fade">
+      <div class="modal-overlay" @click="closeModal" aria-modal="true" role="dialog">
+        <div class="modal-container" @click.stop>
+          <div class="modal-header">
+            <button class="close-button" @click="closeModal" aria-label="Close modal">
+              <span aria-hidden="true">&times;</span>
+            </button>
+          </div>
+          <div class="modal-content">
+            <iframe class="video-iframe"
+              :src="`https://www.youtube.com/embed/${VIDEO_ID}?autoplay=1&modestbranding=1&rel=0`"
+              title="YouTube video player" frameborder="0" allow="autoplay; picture-in-picture"
+              allowfullscreen></iframe>
+          </div>
```

Comment:
Potentially we could also link to the Rolldown Guide in Vite to "give Rolldown a try" (same link as in the video)

Reply:
Done!